### PR TITLE
MBS-8224 Add modified tests instrumentation

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/model/AffectionType.kt [hunk 1]
```diff
@@ -7,7 +7,8 @@ internal enum class AffectionType {
     TEST_MODIFIED,
     TEST_RENAMED,
     /**
-     * Shows that test itself isn't changed, but changed classes in test source sets that are used byt test
+     * Shows that test itself isn't changed, but changed classes in test source sets that are used by test
      */
+    // use?
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/model/AffectionType.kt:12]: What tests shall be that type? Should we run them like modified?

File: subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt [hunk 2]
```diff
@@ -69,6 +69,12 @@ class InstrumentationDefaultConfigPlugin : Plugin<Project> {
                         filters.register("newUiNoE2E") {
                             it.fromSource.includeByAnnotations(TestsFilter.uiNoE2E.annotatedWith)
                         }
+                        filters.register("modifiedUi") {
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt:72]: Should I reduce number of instrumentations? I could merge new and modified tests in one instrumentation

File: subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt [hunk 3]
```diff
@@ -210,6 +225,41 @@ class InstrumentationDefaultConfigPlugin : Plugin<Project> {
         }
     }
 
+    private fun registerModifiedUiConfig(
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt:228]: @dsvoronin Should I reduce the amount of the boilerplate? I stopped by thinking you are moving those pieces to avito

File: subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt [hunk 4]
```diff
@@ -69,6 +69,12 @@ class InstrumentationDefaultConfigPlugin : Plugin<Project> {
                         filters.register("newUiNoE2E") {
                             it.fromSource.includeByAnnotations(TestsFilter.uiNoE2E.annotatedWith)
                         }
+                        filters.register("modifiedUi") {
+                            it.fromSource.includeByAnnotations(TestsFilter.ui.annotatedWith)
+                        }
+                        filters.register("modifiedUiNoE2e") {
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt:75]: I have an idea to create in filter `initWith` method

File: subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt (hunk 2 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt:72]: Different instrumentation tasks give us ability to customize scheduling and quota settings independently.
It seems reasonable to divide these types of changes by instrumentation tasks.

File: subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests-default-config/src/main/kotlin/com/avito/instrumentation/InstrumentationDefaultConfigPlugin.kt:228]: nope, sadly we'll drop all of these

File: subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/model/AffectionType.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/model/AffectionType.kt:12]: That is out of scope for current task

**Review by dsvoronin — APPROVED** [review]

**Review by int02h — APPROVED** [review]

**Pull request merged**